### DISCUSSION TITLE
Understanding Kepler Model Server Benchmark Tests

### DISCUSSION THREAD

**Discussion by husky-parul** [sustainable-computing-io/kepler-model-server#178] [Q&A]
Hi 

I am trying to use KMS benchmarks to collect metrics for PEAKS following this [tutorial](https://github.com/sustainable-computing-io/kepler-model-server/tree/main/model_training#2-run-benchmark-and-collect-metric)

I am running this tests on single node kind cluster which I set up using `./script.sh prepare_cluster`

```
kubectl get nodes -o wide
NAME                              STATUS   ROLES           AGE   VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION           CONTAINER-RUNTIME
kind-for-training-control-plane   Ready    control-plane   19h   v1.25.3   172.18.0.2    <none>        Ubuntu 22.04.1 LTS   6.4.12-200.fc38.x86_64   containerd://1.6.9

```

After that I ran the test benchmark `./script.sh quick_collect`. I have the output in `model_training/data/sample_kepler_query.json`. In the json file I see the following metrics 

![Screenshot from 2023-10-19 08-05-58](https://github.com/sustainable-computing-io/kepler-model-server/assets/24639524/9ce11898-39e4-4b37-ada1-2c90fafc6a27)

 I am interested in capturing the total power consumption of each of the nodes in the cluster. For this which metrics should I use from the following. Also where can I find more information on the relationship/difference between these metrics.
 
 - `kepler_node_core_joules_total` 
- `kepler_node_platform_joules_total`
- `kepler_node_uncore_joules_total`
- `kepler_node_other_joules_total`
- `kepler_node_package_joules_total`

Thanks.

cc @knarayan @sunya-ch @marceloamaral

**Comment by sunya-ch** [comment]
I would suggest to start from `kepler_node_package_joules_total`. In the most system, core and uncore are not available. 

Simple formula is
platform = package + dram + gpu + other (https://github.com/sustainable-computing-io/kepler/blob/63fb1c4434cb693642fb7b024900dcdfda101a6a/pkg/collector/metric/node_metric.go#L422)

rapl reports package, core, uncore, and dram
nvml reports gpu
acpi reports platform
other is computed from the above formula. 

You may check metric description here: https://sustainable-computing.io/design/metrics/#kepler-metrics-for-node-information.
We may need to enrich the explanation if it is unclear.
